commons/src/main/scala/com/mesosphere/usi/async/Retry.scala

```diff
@@ -0,0 +1,146 @@
+package com.mesosphere.usi.async
+
+import java.time.Instant
+import java.time
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.concurrent.{ExecutionContext, Future, Promise, blocking => blockingCall}
+import scala.util.{Failure, Random, Success}
+import scala.util.control.NonFatal
+
+case class RetryConfig(
+    maxAttempts: Int = Retry.DefaultMaxAttempts,
+    minDelay: Duration = Retry.DefaultMinDelay,
+    maxDelay: Duration = Retry.DefaultMaxDelay,
+    maxDuration: Duration = Retry.DefaultMaxDuration
+)
+
+object RetryConfig {
+  def apply(config: Config): RetryConfig = {
+    RetryConfig(
+      config.int("max-attempts", default = Retry.DefaultMaxAttempts), // TODO: should we import RichConfig?
```

Comment:
I wonder if we should import https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/util/RichConfig.scala